prometheus_client/values.py

```diff
@@ -64,7 +65,7 @@ def __reset(self):
                 file_prefix = typ
             if file_prefix not in files:
                 filename = os.path.join(
-                    os.environ['prometheus_multiproc_dir'],
+                    os.environ.get('PROMETHEUS_MULTIPROC_DIR'),
```

Comment:
For users that set the environment variable at runtime with something like `os.environ.setdefault('prometheus_multiproc_dir', '/tmp/multiproc')` will only checking the upper case version here cause their code to stop working?

Looking through github I see that people do sometimes specifically set the ValueClass to MultiProcessValue rather than use get_value_class.

Reply:
Good you looked at that. I'll push a fix to cover that use case too.

Reply:
I updated the PR but now I'm struggling to write a correct set of tests. Can you help me on what?

Reply:
Thanks for updating the PR, I think it looks reasonable but will do one more round of testing after getting the tests working.

It looks like the issue in the tests is that the `__init__` function for an `MmapedValue` is not actually called until a metric is created. That means in each of your tests you will need to create a registry, collector, and perhaps a counter like the other tests below.

Here is a patch that seems to work for me, but double check that it looks sane to you too. Plus you might be able to get rid of some of the duplication.

```diff
diff --git a/tests/test_multiprocess.py b/tests/test_multiprocess.py
index a2a1dbf..845cc3d 100644
--- a/tests/test_multiprocess.py
+++ b/tests/test_multiprocess.py
@@ -32,12 +32,17 @@ class TestMultiProcessDeprecation(unittest.TestCase):
     def tearDown(self):
         del os.environ['prometheus_multiproc_dir']
         del os.environ['PROMETHEUS_MULTIPROC_DIR']
+        values.ValueClass = MutexValue
         shutil.rmtree(self.tempdir)

     def test_deprecation_warning(self):
         os.environ['prometheus_multiproc_dir'] = self.tempdir
         with warnings.catch_warnings(record=True) as w:
-            get_value_class()
+            values.ValueClass = get_value_class()
+            registry = CollectorRegistry()
+            collector = MultiProcessCollector(registry, self.tempdir)
+            Counter('c', 'help', registry=None)
+
             assert os.environ['PROMETHEUS_MULTIPROC_DIR'] == self.tempdir
             assert len(w) == 1
             assert issubclass(w[-1].category, DeprecationWarning)
@@ -47,8 +52,12 @@ class TestMultiProcessDeprecation(unittest.TestCase):
         os.environ['prometheus_multiproc_dir'] = 'should not be picked'
         os.environ['PROMETHEUS_MULTIPROC_DIR'] = self.tempdir
         with warnings.catch_warnings(record=True) as w:
-            get_value_class()
-            assert len(w) == 0
+            values.ValueClass = get_value_class()
+            registry = CollectorRegistry()
+            collector = MultiProcessCollector(registry, self.tempdir)
+            Counter('c', 'help', registry=None)
+
+            assert len(w) == 1


 class TestMultiProcess(unittest.TestCase):
```

Reply:
Thanks @csmarchbanks , tests are passing now.